relstorage/zodbconvert.py

```diff
@@ -105,33 +105,29 @@ def main(argv=sys.argv):
     log.info("Storages opened successfully.")
 
     if options.incremental:
-        if not hasattr(destination, '_adapter') and not hasattr(destination, 'lastTransaction'):
+        if not hasattr(destination, 'lastTransaction'):
             msg = ("Error: no API is known for determining the last committed "
                    "transaction of the destination storage. Aborting "
                    "conversion.")
             sys.exit(msg)
         if not storage_has_data(destination):
             log.warning("Destination empty, start conversion from the beginning.")
         else:
-            wrap_source = False
-            if hasattr(destination, '_adapter'):
-                # RelStorage. Note that we implement lastTransaction(), but
-                # only in-memory, local to the particular object. (We should probably
-                # change that?) So order matters.
-                destination.load(z64) # prime the connection
-                wrap_source = True # compensate for our bug
-                last_tid = destination._adapter.txncontrol.get_tid(
-                    destination._load_cursor)
-            else:
-                # IStorage, like FileStorage
-                last_tid_s = destination.lastTransaction()
-                last_tid = u64(last_tid_s)
+            # First, make sure the connection is fully connected. This
+            # matters for ClientStorage (connects async) and RelStorage
+            # (connects on demand).
+            destination.load(z64)
```

Comment:
It turns out that an empty destination is detected above this and skips the whole incremental processing.